### PR TITLE
statistics: library statistics charts

### PR DESCRIPTION
**Pull request by zannkukai**
* Creates a module for statistic charts.
* Creates library circulation operations chart + admin widget/
* Creates document type repartition chart.

## Dependencies

* https://github.com/rero/rero-ils/pull/3243

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

File: projects/admin/src/environments/environment.ts [hunk 1]
```diff
@@ -30,7 +30,8 @@ export const environment = {
     '/assets/rero-ils-ui/admin/i18n/${lang}.json',
     '/api/translations/${lang}.json'
   ],
-  librarySwitchCheckParamsUrl: ['new', 'edit']
+  librarySwitchCheckParamsUrl: ['new', 'edit'],
+  ngxChartsColorScheme: 'nightLights'  //https://github.com/swimlane/ngx-charts/blob/master/projects/swimlane/ngx-charts/src/lib/utils/color-sets.ts
```
Comment (lauren-d) [inline projects/admin/src/environments/environment.ts:34]: You need to apply this change in 'environment.prod.ts' too

**Review by lauren-d — CHANGES_REQUESTED** [review]

File: projects/admin/src/app/widgets/statistics-charts/utils.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+/**
+ * Format a `Date` object using the YYYY-MM-DD (timezone agnostic).
+ * @param date: the date to parse.
+ * @returns: the formatted date as a string.
+ */
+function _formatDate(date: Date): string {
+  const year = date.toLocaleString("default", { year: "numeric" });
+  const month = date.toLocaleString("default", { month: "2-digit" });
+  const day = date.toLocaleString("default", { day: "2-digit" });
+  return year + '-' + month + '-' + day;
```
Comment (Garfield-fr) [inline projects/admin/src/app/widgets/statistics-charts/utils.ts:28]: suggestion: `[year, month, day].join('-')`

File: projects/admin/src/app/widgets/statistics-charts/components/library/circulation-rate-operation-chart/circulation-rate-operation-chart.component.ts [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    return this._statService
+      .getCirculationRateStatistics(this.libraryPid, this.settings.statisticsFilters)
+      .pipe(
+        map((stats) => {
+          stats.map((serie) => {
+            // this.randomOperationSeries(serie.series);
+            return serie;
+          });
+          return stats;
+        })  // TODO :: remove sample data to debug
```
Comment (Garfield-fr) [inline projects/admin/src/app/widgets/statistics-charts/components/library/circulation-rate-operation-chart/circulation-rate-operation-chart.component.ts:116]: xxxApiService ?

File: projects/admin/src/app/widgets/statistics-charts/components/library/circulation-operation-chart/circulation-operation-chart.component.ts [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        this.settings = settings;
+        this._enableAutoRefresh();
+        this._getStatistics();
+      });
+  }
+
+  /** Return observable used to get statistics for this widget */
+  loadStatistics(): Observable<any> {
+    return this._statService
+      .getCirculationStatistics(this.libraryPid, this.settings.statisticsFilters);
```
Comment (Garfield-fr) [inline projects/admin/src/app/widgets/statistics-charts/components/library/circulation-operation-chart/circulation-operation-chart.component.ts:100]: xxxApiService ?

File: projects/admin/src/app/widgets/statistics-charts/components/library/document-repartition-chart/document-repartition-chart.component.ts [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    private _modalService: BsModalService,
+    private _translateService: TranslateService
+  ) {
+    super();
+  }
+
+  // COMPONENT FUNCTIONS ======================================================
+  /** Get the statistics for this widget */
+  protected loadStatistics(): Observable<any> {
+    return this._statService.getDocTypeRepartitionStatistics(this.libraryPid);
```
Comment (Garfield-fr) [inline projects/admin/src/app/widgets/statistics-charts/components/library/document-repartition-chart/document-repartition-chart.component.ts:75]: xxxApiService ?

**Comment by ManaDeweerdt** [comment]
- [ ] cannot select a date clicking on the calendar icon (but can change the date in the box > ok).

**Pull request closed without merging**